test/e2e-v2/cases/satellite/native-protocols/docker-compose.yml

```diff
@@ -0,0 +1,111 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+version: '2.1'
+
+services:
+
+  etcd:
+    hostname: etcd
+    image: quay.io/coreos/etcd:v3.5.0
+    ports:
+      - 2379
+    networks:
+      - e2e
+    environment:
+      ETCD_LISTEN_CLIENT_URLS: http://0.0.0.0:2379
+      ETCD_ADVERTISE_CLIENT_URLS: http://0.0.0.0:2379
+      ETCD_LISTEN_PEER_URLS: http://0.0.0.0:2380
+      ETCD_INITIAL_ADVERTISE_PEER_URLS: http://0.0.0.0:2380
+      ETCD_INITIAL_CLUSTER: s1=http://0.0.0.0:2380
+      ETCD_NAME: s1
+      ETCD_DATA_DIR: /etcd-data
+    healthcheck:
+      test: ["CMD", "sh", "-c", "etcdctl endpoint health" ]
+      interval: 5s
+      timeout: 60s
+      retries: 120
+
+  oap:
+    extends:
+      file: ../../../script/docker-compose/base-compose.yml
+      service: oap
+    environment:
+      SW_LOG_LAL_FILES: test
+      SW_CLUSTER: etcd
+      SW_METER_ANALYZER_ACTIVE_FILES: spring-sleuth
+      SW_CONFIGURATION: etcd
+    volumes:
+      - ./lal.yaml:/skywalking/config/lal/test.yaml
+    ports:
+      - 12800
+    depends_on:
+      etcd:
+        condition: service_healthy
+
+  satellite:
+    build:
+      context: ../
+      dockerfile: Dockerfile.satellite
+    expose:
+      - 11800
+    ports:
```

Comment:
This is not needed

Reply:
Currently, the `skywalking-infra-e2e` using the `ports` attribute to expose all the ports(export to the environment). And the `ports` means exposed to the host machine. but the `expose` only mean not being exposed to host machines, only exposed to other services.

Reply:
> Currently, the `skywalking-infra-e2e` using the `ports` attribute to expose all the ports(export to the environment). And the `ports` means exposed to the host machine. but the `expose` only mean not being exposed to host machines, only exposed to other services.

That's the point. You should only expose the ports that will be access in local machine to avoid port conflicts in local machine, I think only OAP 12800 is the case. Others like satellite, provider, consumer are all not accessed in local machine and should not be `ports`.

Reply:
Remove the satellite `ports` defines. `etcd`, `provider`, `consumer` and `oap` services need to be exposed to the local machine, so they need `ports` defines.